abstract_domains/state.py

```diff
@@ -1,53 +1,28 @@
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 from abstract_domains.lattice import Lattice
 from copy import deepcopy
 from core.expressions import Expression, VariableIdentifier
 from typing import Set
 
 
-class State(Lattice):
-    class Internal(Lattice.Internal):
-        def __init__(self, kind: Lattice.Kind):
-            """Analysis state internal representation.
-            
-            :param kind: kind of lattice element
-            """
-            super().__init__(kind)
-            self._result = set()    # set of expressions representing the result of the previously analyze statement
-
-        @property
-        def result(self):
-            return self._result
-
-        @result.setter
-        def result(self, result: Set[Expression]):
-            self._result = result
-
-    def __init__(self, kind: Lattice.Kind = Lattice.Kind.Default):
+class State(Lattice, ABC):
```

Comment:
Lattice is now the default WITH Top and Bottom element. If somebody do not want the this, they can still subclass BaseLattice.